core/arch/arm/include/sm/optee_smc.h

```diff
@@ -93,6 +93,12 @@
 #define OPTEE_SMC_SHM_CACHED		1
 
 /*
+ * Privileged call flag. Used by supplicant. Asks OP-TEE to use reserved
+ * thread if all regular threads are used.
+ */
+#define OPTEE_SMC_FLAGS_PRIVILEGED	BIT(0)
```

Comment:
Okay, point taken. Generally speaking, RPC handling code should not issue OP-TEE calls because this calls can lead to more RPCs.

But in this particular situation with SHM we can:

a) make exception for special `register_shm` call
b) rework the way how results are returned from RPC

Variant a) leads to alterations to tasker/scheduler code. Variant b) leads to more complex ABI between OP-TEE,  kernel driver and supplicant. I prefer b) because I did it once.

What do you think?